pkg/api/store/globaldns/globaldns_store.go

```diff
@@ -0,0 +1,86 @@
+package globaldns
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	managementv3 "github.com/rancher/types/client/management/v3"
+)
+
+func Wrap(store types.Store) types.Store {
+	storeWrapped := &Store{
+		Store: store,
+	}
+	storeWrapped.mu = sync.Mutex{}
+	return storeWrapped
+}
+
+type Store struct {
+	types.Store
+	mu sync.Mutex
+}
+
+func (p *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+	fqdn := convert.ToString(data["fqdn"])
+	if fqdn == "" {
+		return nil, httperror.NewFieldAPIError(httperror.MissingRequired, "FQDN", "")
+	}
+
+	p.mu.Lock()
```

Comment:
Just curious about why this mutex lock/unlock is required here?

Reply:
A guard against concurrent creates/updates - as much as possible - to avoid clashes.